src/generator.cs

```diff
@@ -707,6 +707,34 @@ public DelegateNameAttribute (string s)
 	public string Name { get; set; }
 }
 
+//
+// Used to specify the delegate property name that will be created when
+// the generator creates the delegate property on the host
+// class that holds events.
+//
+// This is really useful when you have two overload methods that makes
+// sense to keep them named as is but you want to expose them in the host class
+// with a better given name.
+//
+// example:
+// interface SomeDelegate {
+//     [Export ("foo"), DelegateApiName ("Confirmation"), DelegateName ("Func<bool>"), DefaultValue (false)]
+//     bool Confirm (Some source);
+// }
+//
+// Generates propety in the host class:
+//	Func<bool> Confirmation { get; set; }
+//
+//
+public class DelegateApiNameAttribute : Attribute {
```

Comment:
Done in https://github.com/xamarin/xamarin-macios/pull/817/commits/e5d584e8bf0c01bbcd0795b2e1a431cc6f6b1c1e